cdap-data-fabric/src/main/java/co/cask/cdap/data2/nosql/NoSqlStructuredTableRegistry.java

```diff
@@ -128,6 +128,15 @@ public void registerSpecification(StructuredTableSpecification specification) th
   @Override
   public StructuredTableSpecification getSpecification(StructuredTableId tableId) {
     Optional<StructuredTableSpecification> optional = specCache.getUnchecked(tableId);
+    if (optional.isPresent()) {
+      return optional.get();
+    }
+    // If spec is not available in cache, then check storage to see if the table is now created
+    optional = getSpecificationFromStorage(tableId);
```

Comment:
Isn't this already being done internally when you call specCache.getUnchecked (on line 130)?

Reply:
Ah, the cache might be caching the initial null value

Reply:
It seems like a bad idea/pattern to have the cache hold values of type Optional, then.

Reply:
The loading cache does not allow not caching null values, so trying to workaround that limitation.

Reply:
I also don’t see a simple solution to this.
LGTM